pkg/datastore/credentials.go

```diff
@@ -0,0 +1,85 @@
+package datastore
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	rdsauth "github.com/aws/aws-sdk-go-v2/feature/rds/auth"
+	"golang.org/x/exp/maps"
+
+	log "github.com/authzed/spicedb/internal/logging"
+)
+
+// CredentialsProvider allows datastore credentials to be retrieved dynamically
+type CredentialsProvider interface {
+	// Name return the name of the provider
+	Name() string
+	// Get return the username and password to use when connecting to the underlying datastore
+	Get(ctx context.Context, dbHostname string, dbPort uint16, dbUser string) (string, string, error)
+}
+
+var NoCredentialsProvider CredentialsProvider = nil
```

Comment:
`const` (if it lets you)

Reply:
no worky `pkg/datastore/credentials.go:24:29: invalid constant type CredentialsProvider`